docs/reference/modules/node.asciidoc

```diff
@@ -32,42 +33,45 @@ configure this setting, the node has the following roles by default:
 * `remote_cluster_client`
 * `transform`
 
-NOTE: If you set `node.roles`, the node is assigned only the roles you specify.
+IMPORTANT: If you set `node.roles`, ensure you specify every node role your
+cluster needs. For example, to use {transforms}, you must specify the
+`transform` role for one or more nodes in your cluster. For resiliency,
+production {es} clusters require an elected `master` node and at least one node
+for each role. See <<high-availability-cluster-design>>.
```

Comment:
"For resiliency" we will need more than "at least one node for each role". Resiliency requires redundancy, so this final sentence isn't quite right however the link to the HA cluster page is useful.

I personally prefer the concept of naming which roles are most likely to be needed. For example `...A minimal cluster requires at least one master, data, ingest and transform node. For production, resilience is required ...`.

The reason why I like naming the roles, is that a user may not know they are using a transform, because it was installed as part of endpoint. They may have no requirement to use transforms as far as they are aware, but are still using them.

Reply:
:doh: Thanks for catching my error. I obviously got my requirements mixed up. 😬  Now that I better understand your feedback, I'm not sure if we should include that section anyway.

I think the definition of a `minimal` cluster will likely differ based on use case, but I agree that we should highlight which roles are needed to access most Stack features. 

I've revised the admonition based on your suggestion and tested locally to ensure our solutions still stay up. As you mentioned offline, I also feel that this should be addressed in the UI.

As this now falls into the area of a "best practice," I'll also get a review from @jasontedor. Thanks again, @sophiec20!